src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENestedNamespaceSymbol.cs

```diff
@@ -125,11 +125,14 @@ internal override ModuleSymbol ContainingModule
 
         protected override void EnsureAllMembersLoaded()
         {
-            var typesByNS = _typesByNS;
+            var typesByNS = Volatile.Read(ref _typesByNS);
 
-            if (lazyTypes == null || lazyNamespaces == null)
+            if (typesByNS == null)
+            {
+                Debug.Assert(lazyNamespaces != null && lazyTypes != null);
+            }
+            else
             {
-                System.Diagnostics.Debug.Assert(typesByNS != null);
```

Comment:
This was a simple thread-timing issue waiting to happen. Rather than using `lazyTypes` and `lazyNamespaces` as proxies to the real sentinel, we should just use `_typesByNS`, which is the actual sentinel that work was completed.